src/ic_renderer.h

```diff
@@ -16,13 +19,19 @@ namespace IC {
 
     class Renderer {
     public:
-        Renderer(RendererConfig config) { window = config.window; };
+        Renderer(RendererConfig config) { window = config.window; }
         ~Renderer() = default;
 
-        virtual void DrawFrame() = 0;
+        void AddImguiFunction(ImGuiFunction function) { imGuiFunctions.push_back(function); }
+        void RemoveImguiFunction(ImGuiFunction function) {
+            imGuiFunctions.erase(std::find(imGuiFunctions.begin(), imGuiFunctions.end(), function));
```

Comment:
Does the order of the imGuiFunctions vector matter? If not can we use [std::erase](https://en.cppreference.com/w/cpp/container/vector/erase2) instead? It'll be more efficient here

Reply:
I don't think order matters? I switched to erase instead

thanks, I figured there was probably a better way to do this lol